### DISCUSSION TITLE
Extending Language Configuration

### DISCUSSION THREAD

**Discussion by tim-hilt** [microsoft/vscode-discussions#411] [Extension Development QnA] [answered]
I have multiple questions:

- Can I extend an existing language-configuration or does it override the existing one?
- Can a language-configuration be defined for multiple languages at the same time?
- Do I need to write a VS Code extension to add/extend a language-configuration or can I also define it locally?

**Goal/Rationale**

When I open a C++ file and type `/*` it will be completed to `/*|*/` with `|` being the cursor. When I now hit enter, it will look like this:

``` cpp
/*
|*/
```

but what I want it to look like is this:

``` cpp
/*
 * |
 */
```

this would basically be useful for all languages, that have this block-comment style.

**Comment by isidorn** [comment]
I think this is in Core, and might not be contributable, I am not sure but @hediet might know.

**Reply by tim-hilt** [reply, marked as answer]
I have since found out about https://github.com/kevb34ns/auto-comment-blocks which does exactly that: define a language-configuration for multiple languages. It only defines comment-related stuff though. Just tried it out and the rest of the language-config still seems to work.

Auto Comment Blocks doesn’t do what I want though, this discussion therefore doesn’t have a solution yet.
